qiskit/transpiler/passes/layout/dense_layout.py

```diff
@@ -79,6 +79,11 @@ def run(self, dag):
             raise TranspilerError(
                 "A coupling_map or target with constrained qargs is necessary to run the pass."
             )
+        if not self.coupling_map.is_connected():
+            raise TranspilerError(
+                "Coupling Map is disjoint, this pass can't be used with a disconnected coupling "
+                "map."
+            )
```

Comment:
Does `DenseLayout` actually fail if the number of qubits fits in at least one of the individual components?

Reply:
I was thinking that it probably wouldn't fail would potentially pick an invalid subgraph because the algorithm looks for the subgraph of size n (where n is circuit width) with the highest degree of connectivity. But upon reflection yeah you're right this actually would only fail if the circuit width is less than `CouplingMap.size()` but greater than `CouplingMap.largest_connected_component()` because the bfs wouldn't find a subgraph large enough.

I was just proactively trying to block against people trying to use disjoint coupling maps with `transpile()` until we add proper support to map components from the circuit to the target in layout and routing. But in the meantime I can adjust this check to be:

```suggestion
        if dag.size() > self.coupling_map.largest_connected_component():
            raise TranspilerError(
                "Coupling Map is disjoint, this pass can't be used with a disconnected coupling "
                "map for a circuit this large.."
            )
```

Reply:
(This question doesn't actually matter, in retrospect - we can do it in a follow-up whatever the answer.)

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/9710/commits/6b565170acd687fb8d3adba75f9097537281f768